internal/framework/services/s3/bucket_policy_resource.go

```diff
@@ -57,10 +57,12 @@ type bucketPolicyStatement struct {
 	Effect    string                          `json:"Effect"`
 	Action    []string                        `json:"Action"`
 	Resources []string                        `json:"Resource"`
-	Principal []string                        `json:"Principal"`
+	Principal *Principal                      `json:"Principal"`
 	Condition *bucketPolicyStatementCondition `json:"Condition,omitempty"`
 }
-
+type Principal struct {
```

Comment:
type doesn't need to be exported

Reply:
reverted to a slice of strings. it's easier.